tools/libfrencutils/create_xgrid.c

```diff
@@ -2100,13 +2099,57 @@ double poly_ctrlat(const double x[], const double y[], int n)
     avg_y = (lat1+lat2)*0.5;
     if      (dx==0.0) continue;
     if(dx > M_PI)  dx = dx - 2.0*M_PI;
-    if(dx < -M_PI) dx = dx + 2.0*M_PI;
+    if(dx <= -M_PI) dx = dx + 2.0*M_PI; // flip sign for dx=-pi to fix huge value see comments in function poly_area
```

Comment:
The addition of  "=" sign is the main bug fix for issue #180 . The rest of this update is adding/fixing comments.